lib/limits/checks.js

```diff
@@ -0,0 +1,161 @@
+// Functions to apply limits to nodes
+
+'use strict';
+
+var async = require('async');
+var nodeError = require('../limits/error');
+var estimates = require('../limits/estimates');
+
+// Obtain limit values given the limit definitions, which has the form:
+//   {  maximumNumberOfRows: {          // name of the limit argument as used by checker functions
+//          default: 1000000,           // limit default value
+//          message: 'too many rows'    // error to be shown when limit is exceeded
+//          name: 'maximumNumberOfRows' // (optional) name of the limit configuration parameter
+//          ...
+// The result if of the form:
+//   {  maximumNumberOfRows: {          // name of the limit argument as used by checker functions
+//          value: 1000000,             // value of the limit
+//          message: 'too many rows'    // error to be shown when limit is exceeded
+//          ...
+function limitValues(node, context, limits, callback) {
+    var values = {};
+    async.forEachOf(limits, function(limitParams, limit, done) {
```

Comment:
No need really, at some point in writing here I was expecting that getting limit values could eventually need some async operation, but now I see no reason for this. I'll remove it. My only doubt is whether to maintain this `limitValues` function as async (with callback) or not...